src/map/lua/lua_baseentity.cpp

```diff
@@ -2668,17 +2668,22 @@ void CLuaBaseEntity::setPos(sol::variadic_args va)
                 return;
             }
 
-            ((CCharEntity*)m_PBaseEntity)->loc.destination = zoneid;
-            ((CCharEntity*)m_PBaseEntity)->status          = STATUS_TYPE::DISAPPEAR;
-            ((CCharEntity*)m_PBaseEntity)->loc.boundary    = 0;
-            ((CCharEntity*)m_PBaseEntity)->m_moghouseID    = 0;
-            ((CCharEntity*)m_PBaseEntity)->clearPacketList();
-            charutils::SendToZone((CCharEntity*)m_PBaseEntity, 2, zoneutils::GetZoneIPP(m_PBaseEntity->loc.destination));
-            //((CCharEntity*)m_PBaseEntity)->loc.zone->DecreaseZoneCounter(((CCharEntity*)m_PBaseEntity));
+            if (static_cast<CCharEntity*>(m_PBaseEntity)->PPet != nullptr)
```

Comment:
Looking at this now, we should probably follow the example used in the upper block, and not cast each time, but instead declare PChar in this one as well (`auto* PChar = static_cast<CCharEntity*>(m_PBaseEntity);`)  That'll reduce the bulk of those repeated casts, and just replace with the new var.